src/env.cc

```diff
@@ -857,10 +857,13 @@ void Environment::AsyncHooks::grow_async_ids_stack() {
 uv_key_t Environment::thread_local_env = {};
 
 void Environment::Exit(int exit_code) {
-  if (is_main_thread())
+  if (is_main_thread()) {
+    stop_sub_worker_contexts();
```

Comment:
I was reviewing this method and its callee chains, and have this doubt:
 - `Worker::Exit | Isolate::TerminateExecution` - are these callable from another thread? Of course, it is possible to do so, and any thread is able to modify the data structures pertinent to a worker - but we are doing nothing to stop the execution of the `thread` that is represented by this worker? In other words, how / at what point are we physically stopping the worker thread?

Reply:
> `Worker::Exit | Isolate::TerminateExecution` - are these callable from another thread?

Yes, they are essentially `worker.terminate()` – in particular, `Isolate::TerminateExecution()` is one of the few thread-safe V8 functions.

> In other words, how / at what point are we physically stopping the worker thread?

We’re stopping the event loop and disallowing JS execution, waiting for everything to finish and then join the thread … as far as I could tell, that’s the best we can do here.

I guess this might be problematic when some threads have environment cleanup handlers registered via addons that block indefinitely? That would not really be a bug on our side, though…